WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/UsernameChangerFullScreenDialogFragment.java

```diff
@@ -0,0 +1,322 @@
+package org.wordpress.android.ui.accounts.signup;
+
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.os.Handler;
+import android.support.annotation.Nullable;
+import android.support.design.widget.TextInputEditText;
+import android.support.v4.app.Fragment;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.SimpleItemAnimator;
+import android.text.Editable;
+import android.text.Html;
+import android.text.Spanned;
+import android.text.SpannedString;
+import android.text.TextUtils;
+import android.text.TextWatcher;
+import android.view.ContextThemeWrapper;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+import org.wordpress.android.R;
+import org.wordpress.android.WordPress;
+import org.wordpress.android.analytics.AnalyticsTracker;
+import org.wordpress.android.fluxc.Dispatcher;
+import org.wordpress.android.fluxc.generated.AccountActionBuilder;
+import org.wordpress.android.fluxc.store.AccountStore.FetchUsernameSuggestionsPayload;
+import org.wordpress.android.fluxc.store.AccountStore.OnUsernameSuggestionsFetched;
+import org.wordpress.android.ui.FullScreenDialogFragment.FullScreenDialogContent;
+import org.wordpress.android.ui.FullScreenDialogFragment.FullScreenDialogController;
+import org.wordpress.android.ui.accounts.signup.UsernameChangerRecyclerViewAdapter.OnUsernameSelectedListener;
+import org.wordpress.android.util.ActivityUtils;
+import org.wordpress.android.util.AppLog;
+import org.wordpress.android.util.AppLog.T;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+public class UsernameChangerFullScreenDialogFragment extends Fragment implements
+        FullScreenDialogContent, OnUsernameSelectedListener {
+    private ProgressBar mProgressBar;
+
+    protected FullScreenDialogController mDialogController;
+    protected Handler mGetSuggestionsHandler;
+    protected RecyclerView mUsernameSuggestions;
+    protected Runnable mGetSuggestionsRunnable;
+    protected String mDisplayName;
+    protected String mUsername;
+    protected String mUsernameSelected;
+    protected String mUsernameSuggestionInput;
+    protected TextInputEditText mUsernameView;
+    protected TextView mHeaderView;
+    protected UsernameChangerRecyclerViewAdapter mUsernamesAdapter;
+    protected boolean mShouldWatchText;  // Flag handling text watcher to avoid network call on device rotation.
+    protected int mUsernameSelectedIndex;
+
+    public static final String EXTRA_DISPLAY = "EXTRA_DISPLAY";
```

Comment:
I guess this one was meant to be named `EXTRA_DISPLAY_NAME` instead, right?

Reply:
I updated the constant name in 7b63b4316c3a2fb4f633c64f9f16f31819baca91.